core/base_abstractions/sensor.py

```diff
@@ -158,23 +159,60 @@ def _get_observation_space(self) -> gym.spaces.Tuple:
         only if the expert failed to generate a true expert action. The
         value `num actions in task` should be in `config["nactions"]`
         """
-        return gym.spaces.Tuple(
-            (gym.spaces.Discrete(self.nactions), gym.spaces.Discrete(2))
-        )
+        return gym.spaces.Tuple((self.action_space, gym.spaces.Discrete(2)))
 
     def get_observation(
         self, env: EnvType, task: SubTaskType, *args: Any, **kwargs: Any
     ) -> Any:
         # If the task is completed, we needn't (perhaps can't) find the expert
         # action from the (current) terminal state.
         if task.is_done():
-            return np.array([-1, False], dtype=np.int64)
+            return None, False
```

Comment:
The changes in this file are completely broken (I see there was a copy-paste error that further broke things here. Ignote this file for now).

Reply:
I've tried to make a bit more sense of this. Currently I'm assuming the tasks' query_expert methods will return either a gym-spaces-flattened action and a bool OR an int and a bool. The Sensor always provides a flattened observation composed of the custom-flattened action and the bool flag. It's easier to understand in code than with this description :)